### PR TITLE
[#240] Intercept ClientException when iterating over statement keys

### PR DESCRIPTION
**Pull request by fbiville**

### CODE REVIEW

**Comment by fbiville** [comment]
~Requires https://github.com/neo4j-contrib/neo4j-jdbc/pull/242 to be merged.~
Edit: DONE.

**Comment by conker84** [comment]
@fbiville can you please rebase in order to add the change of #242 in this one?

**Comment by fbiville** [comment]
@conker84 done, thanks!

File: neo4j-jdbc-bolt/src/main/java/org/neo4j/jdbc/bolt/BoltNeo4jResultSet.java [hunk 1]
```diff
@@ -91,8 +92,14 @@ private BoltNeo4jResultSet(Statement statement, StatementResult iterator, int...
...
-			//Keys are exactly the ones returned from the iterator
-			this.keys = this.iterator.keys();
+			// Keys are exactly the ones returned from the iterator
+			// This may raise an exception if the statement is invalid
+			try {
+				this.keys = this.iterator.keys();
+			}
+			catch (ClientException ce) {
+				throw new SQLException(ce.getMessage(), ce);
+			}
```
Comment (conker84) [inline neo4j-jdbc-bolt/src/main/java/org/neo4j/jdbc/bolt/BoltNeo4jResultSet.java]: I would move the try-catch at line 89 wrapping the whole `if/else/if` block. WDYT?

**Review by conker84 — COMMENTED** [review]
@fbiville thank you so much, please look at my comment

File: neo4j-jdbc-bolt/src/main/java/org/neo4j/jdbc/bolt/BoltNeo4jResultSet.java (hunk 1 above)
Comment (fbiville) [reply neo4j-jdbc-bolt/src/main/java/org/neo4j/jdbc/bolt/BoltNeo4jResultSet.java]: that's probably safer indeed :)
let me try it

**Comment by conker84** [comment]
@fbiville thank you!!!

**Pull request merged**